image-hd.c

```diff
@@ -275,7 +283,8 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 		/* reserve space for extended boot record if necessary */
 		if (part->in_partition_table)
 			++partition_table_entries;
-		if (has_extended && (partition_table_entries > 3))
+		if (has_extended &&
+			(partition_table_entries > (hd->primary_partitions-1)))
 			part->extended = cfg_true;
```

Comment:
The indention makes this hard to read. Maybe this:
```
part->extended = has_extended &&
	(partition_table_entries > (hd->primary_partitions - 1));```

Or something like that depending on other changes.

Reply:
okay